subsys/net/lib/sockets/Kconfig

```diff
@@ -100,6 +100,15 @@ config NET_SOCKETS_TLS_MAX_CIPHERSUITES
 	  By default, all ciphersuites that are available in the system are
 	  available to the socket.
 
+config NET_SOCKETS_TLS_MAX_APP_PROTOCOLS
+	int "Maximum number of supported application layer protocols"
+	default 3
+	depends on NET_SOCKETS_SOCKOPT_TLS
```

Comment:
Shouldn't we also depend on `MBEDTLS_SSL_ALPN` here?
Why 3 as a default value, would 1 be enough in normal case? What kind of protocols are set in the ALPN list?

Reply:
Yes. "h2", "http/1.1" are common ones for http2/1.1 support.

Reply:
Perhaps,

```
default 2 if MBEDTLS_SSL_ALPN
default 0
```

Reply:
Sure, that is one option. You just then need to tweak the
`const char *alpn_list[CONFIG_NET_SOCKETS_TLS_MAX_APP_PROTOCOLS + 1];`
as it has the +1 in it.